kite-data/kite-data-core/src/main/java/org/kitesdk/data/PartitionStrategy.java

```diff
@@ -160,6 +163,56 @@ PartitionStrategy getSubpartitionStrategy(int startIndex) {
         fieldPartitioners.size()));
   }
 
+  public void checkCompatibleUpdate(PartitionStrategy other, Schema schema) {
```

Comment:
This shouldn't really be exposed as a public method. Put it on a class in the SPI?